pkg/scheduler/scheduler.go

```diff
@@ -440,18 +440,19 @@ func (sched *Scheduler) scheduleOne() {
 	// Synchronously attempt to find a fit for the pod.
 	start := time.Now()
 	suggestedHost, err := sched.schedule(pod)
-	metrics.SchedulingAlgorithmLatency.Observe(metrics.SinceInMicroseconds(start))
 	if err != nil {
 		// schedule() may have failed because the pod would not fit on any host, so we try to
 		// preempt, with the expectation that the next time the pod is tried for scheduling it
 		// will fit due to the preemption. It is also possible that a different pod will schedule
 		// into the resources that were preempted, but this is harmless.
 		if fitError, ok := err.(*core.FitError); ok {
+			preemptionStartTime := time.Now()
 			sched.preempt(pod, fitError)
+			metrics.SchedulingAlgorithmPremptionEvaluationDuration.Observe(metrics.SinceInMicroseconds(preemptionStartTime))
 		}
 		return
 	}
-
+	metrics.SchedulingAlgorithmLatency.Observe(metrics.SinceInMicroseconds(start))
```

Comment:
The original issue is about: Add metrics for **number** of preemptions and **number** of victims.` Not the time it consumed (also the preemption time does not matter this much).

You may want to rework the PR :smiley: I would like to review then.

Reply:
Sure @resouer - Thanks for pointing it, I saw it after I submitted my PR. I have already started working on it. But I think preemption time is important as it has certain computation involved and we need measure this as well if the cluster is fully loaded and frequent preemptions are happening.

Reply:
Sure, I am totally positive with this as long as we are now on the right track